requirements.dev.txt

```diff
@@ -4,4 +4,5 @@ mypy-extensions==1.0.0
 pytest==7.4.4
 pytest-asyncio==0.23.3
 pre-commit==3.5.0
-ruff==0.2.0
+ruff==0.7.1
+console-ctrl==0.1.0
```

Comment:
This looks like it's about 15 LoC total; maybe we should just vendor it?

Reply:
We could, but it's also only used for tests and not a lib dependency

Reply:
Yeah — don't think it matters too much either way, just seems like more of a snippet than a library :D